### ISSUE TITLE
Plans for tesseract 5.x.y

### ISSUE THREAD

**Issue by amitdo** [tesseract-ocr/tesseract#3673]
I suggest to focus on 5.x for 2022 at least.

That means we should not break the API (and ABI?). Use C++17, not C++20/C++23.

**Comment by stweil** [comment]
What about releasing a 5.0.1 after Christmas at the end of December? I think there are several fixes since 5.0.0 which would be good for a new release.

**Comment by amitdo** [comment]
Mind reader :-)
I was about to suggest to release 5.0.1 before year end. But it would be nice if we can fix #3683 before releasing 5.0.1.

**Comment by amitdo** [comment]
Right before tagging 5.0.1. you can update this sentence from the README:

>The latest stable version is 5.0.0, released on November 30, 2021.

**Comment by egorpugin** [comment]
What should be added into v5?
5.x changes could be merged into branch and cherry picked into v6 main.

**Comment by stweil** [comment]
We already have a wish list for improved training, a lot of issues with layout detection, want improved logging, and much more. Maintaining two branches did not work good with 4.x, and I am afraid it would not work better with 5.x.

**Comment by egorpugin** [comment]
Maybe keep 5.0 as is? It is a good release with a number of changes.
Everything else will go straight into 6?

**Comment by amitdo** [comment]
https://github.com/tesseract-ocr/tessdoc/blob/main/ReleaseNotes.md#changes-made-since-last-release

**Comment by amitdo** [comment]
> What about releasing a 5.0.1 after Christmas at the end of December? I think there are several fixes since 5.0.0 which would be good for a new release.

Do you plan to release 5.0.1 next week?

**Comment by stweil** [comment]
Yes, unless we discover that something very important is still missing.

**Comment by stweil** [comment]
> It would be nice if we can fix #3683 before releasing 5.0.1.

There is still no fix, and I still have no `clang-cl`, so I cannot look for a fix myself. Should we release 5.0.1 without a fix? Are other things missing for 5.0.1 (besides updating of the documentation)?

**Comment by egorpugin** [comment]
clang-cl is not worth it currently.

**Comment by amitdo** [comment]
You can release 5.0.1 without the clang-cl fix.

**Comment by stweil** [comment]
Release 5.0.1 is now online.

**Comment by stweil** [comment]
The next release could be a new minor version 5.1.0 with new features, maybe end of January (unless there is an urgent need for a bug fix release 5.0.2). I want to have especially image information in ALTO and hOCR output (see PR #3710 which implements that for hOCR), maybe more from the [project list](https://github.com/tesseract-ocr/tesseract/projects/1).

**Comment by amitdo** [comment]
https://packages.ubuntu.com/search?keywords=tesseract-ocr

@AlexanderP,

Are you going to update Ubuntu 22.04 to 5.0.1 soon? The feature freeze date is February 24.

**Comment by AlexanderP** [comment]
@amitdo 

i uploaded:
- tesseract-ocr 4.1.3 and 5.0.1  https://mentors.debian.net/package/tesseract
- tesseract-lang 1:5.0.0~git39-6572757-1 https://mentors.debian.net/package/tesseract-lang/

I hope @jbreiden  will upload them to debian.

**Comment by amitdo** [comment]
Hi @AlexanderP,

>I hope @jbreiden will upload them to debian.

From https://tracker.debian.org/pkg/tesseract :

>maintainer: [Alexander Pozdnyakov]

So, why can't you **directly** push new versions of Tesseract to Debian?

**Comment by stweil** [comment]
I'd like to create a new release Tesseract 5.1.0 soon. Originally I had planned it for end of January.

Are there any contributions or important bug fixes which should be included still pending (then I'd wait), or can we release now?

**Comment by Shreeshrii** [comment]
I suggest you go ahead with 5.1.0 now.

I would like to see improvements related to training and evaluation implemented, but they could go in a future release.

**Comment by stweil** [comment]
[Release 5.1.0](https://github.com/tesseract-ocr/tesseract/releases/tag/5.1.0) is now available.

**Comment by AlexanderP** [comment]
@amitdo no rights to upload to debian

**Comment by stweil** [comment]
There are now several fixes and improvements in git master, so I think it's time for a new release 5.1.1.

@egorpugin, is it possible to fix the CI sw build which is currently failing?

Are there any other contributions or important bug fixes which should be included still pending (then I'd wait), or can we release now?

**Comment by egorpugin** [comment]
Yes, I'll check.

**Comment by zdenop** [comment]
Unfortunately  windows build does not work (for me): I tried Clang (14) and MS Visual Studio (2019). Here are logs:
[clang_build.zip](https://github.com/tesseract-ocr/tesseract/files/8811716/clang_build.zip)
[msvc_build.zip](https://github.com/tesseract-ocr/tesseract/files/8811717/msvc_build.zip)

**Comment by amitdo** [comment]
`cmake-win64` action fails (since March 29).

`cmake` and `vcpkg` actions pass.

**Comment by zdenop** [comment]
@egorpugin: VS2019 is quite heavily used. I would suggest supporting it with the next release...

**Comment by egorpugin** [comment]
I've added vs2019 and ubuntu22.04.
Let's check how they work
https://github.com/tesseract-ocr/tesseract/commit/b3011ef72ce9ba6f27d791d5efdbb970f03fbd3f

**Comment by egorpugin** [comment]
> 22.04 fails: https://github.com/tesseract-ocr/tesseract/runs/6693857859.

Github ci has beta ubuntu22.04. I'll disable it probably for now.

**Comment by amitdo** [comment]
So @stweil, are you going to make a new release in the upcoming days?

**Comment by stweil** [comment]
If that's fine for everybody, I'll do so. Thank you for the latest fixes to all who where involved.

**Comment by zdenop** [comment]
I would like to see solved #3092 - not sure if `pixRemoveColormap` is needed when alpha channel is removed. I also do not like that [api->GetInputImage()](https://github.com/tesseract-ocr/tesseract/blob/424b17f997363670d187f42c43408c472fe55053/src/api/baseapi.cpp#L924) / `tesseract_->pix_original()` returns modified image(without alpha channel, after pixRemoveColormap)...

Also, it would be great to check if [some bug](https://github.com/tesseract-ocr/tesseract/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen+label%3Abug) can not be closed...

**Comment by zdenop** [comment]
Next 7 days I will have no time for the opensource projects, so there will be minimum input from my side. Maybe it is the eptonica issue - let's wait for Dan reaction.

**Comment by amitdo** [comment]
Any bug that won't be fixed before 5.1.1 is release, can still be fixed in the next release(s).

I expect there will be at least one more 5.x.y release this year. @stweil, can you confirm this is your plan?

**Comment by zdenop** [comment]
Just a quick note: When I build example code in #1231 it works with a combination of recent tesseract code + leptonica 1.82.0.
But with recent leptonica code (a.k.a. 1.83.0) it fails:

```cmd
GetComponentImages.cpp(21): error C2027: use of undefined type 'Boxa'
f:/win64/include\tesseract/baseapi.h(37): note: see declaration of 'Boxa'
GetComponentImages.cpp(21): warning C4473: 'printf' : not enough arguments passed for format string
GetComponentImages.cpp(21): note: placeholders and their parameters expect 1 variadic arguments, but 0 were provided
GetComponentImages.cpp(21): note: the missing variadic argument 1 is required by format string '%d'
GetComponentImages.cpp(22): error C2027: use of undefined type 'Boxa'
f:/win64/include\tesseract/baseapi.h(37): note: see declaration of 'Boxa'
GetComponentImages.cpp(24): error C2027: use of undefined type 'Box'
f:\win64\include\leptonica\imageio.h(241): note: see declaration of 'Box'
GetComponentImages.cpp(24): error C2660: 'tesseract::TessBaseAPI::SetRectangle': function does not take 1 arguments
f:/win64/include\tesseract/baseapi.h(340): note: see declaration of 'tesseract::TessBaseAPI::SetRectangle'
GetComponentImages.cpp(29): error C2027: use of undefined type 'Box'
f:\win64\include\leptonica\imageio.h(241): note: see declaration of 'Box'
```

If I got it right we should require for the next release `leptonica <= 1.82`.

**Comment by amitdo** [comment]
This one is Leptonica's issue.

**Comment by amitdo** [comment]
https://github.com/DanBloomberg/leptonica/commit/e8b670d9357c78a6

https://github.com/DanBloomberg/leptonica/commit/29a33a364848f67d0

They didn't do the same for cmake/sw.

BTW, It does not look right to me to install a file with `_internal` in its name.

**Comment by amitdo** [comment]
I think you can solve the compiler error by using forward declaration for Box and Boxa and by using `boxaGetCount()` and `boxGetGeometry()`.

**Comment by amitdo** [comment]
So @zdenop, is it okay by you to move forward and release 5.1.1 next week, after Egor will fix sw?

**Comment by egorpugin** [comment]
It was fixed.
Now I transfer DNS, so website (and binaries from there) will be unavailable for few days.

**Comment by amitdo** [comment]
We can revert this commit if you want, and re-apply it after 5.1.1 is released.

**Comment by stweil** [comment]
For me it does not matter whether a 5.1.1 or a 5.2.0 is released. I'd keep that commit.

**Comment by zdenop** [comment]
What about #3847 (Maybe have a look at other open PRs...)

**Comment by amitdo** [comment]
https://github.com/DanBloomberg/leptonica/issues/629

**Comment by amitdo** [comment]
If someone wants to run performance measurements I have also a [branch](https://github.com/stweil/tesseract/tree/box_internals) which only uses the public Leptonica API in commit https://github.com/stweil/tesseract/commit/9c74918fea37c68958cc2857af93537e66f9a81c.

It's all about font rendering. I don't see a reason not to apply this patch for 5.1.2.

**Comment by amitdo** [comment]
@stweil, can you release 5.2.0 today or in the next few days?

More bug fixes and enhancements can always be done in the next release...

**Comment by stweil** [comment]
[5.2.0](https://github.com/tesseract-ocr/tesseract/releases/tag/5.2.0) is now released. Thank you to all contributors.

**Comment by amitdo** [comment]
Thank you, Stefan.

**Comment by amitdo** [comment]
@zdenop,

Can you please announce the release of Tesseract 5.2.0 in the forum?

**Comment by zdenop** [comment]
done.

**Comment by mars4science** [comment]
Hi all! I've recently started to use Tesseract. Thank you! 

I have some ideas on improvements, I've looked at releases and have not found my issues resolved (I'm on 4.1.1 from my distro Linux Mint).

Documentation: it takes some trial and error and googling to understand basics. For ordinary open source user, IMO basic info better be in man pages. Basic for me was: 
1) uniformly colored background (when I added white margins to grey photo, recognition started to return close to nothing).
2) left/right empty margins needed (when I supplied image where text started right at edge, the result omitted leftmost several symbols)
3) black and white better than grey: I've used GIMP Color-Exposure to convert.
Maybe some other basic tips to include?

Now to something not manual pages related. I've noted sometimes output contained fF and 1l for f and 1 respectively. I understand tesseract might be trying to give both likely symbols, but how about informing user of those few places where recognition was not sure?

TIA

**Comment by amitdo** [comment]
@stweil,

Do you think we have enough changes to justify a new minor release? Maybe we can release 5.3.0 before the year's end. 

https://github.com/tesseract-ocr/tessdoc/blob/main/ReleaseNotes.md

**Comment by stweil** [comment]
Yes, I think so, too. @all, are there open issues or pull requests which should be included in 5.3.0?

**Comment by zdenop** [comment]
Outside of the already open issues: 
- clang on windows reports several  set but not used  variables [clang_warnings.txt](https://github.com/tesseract-ocr/tesseract/files/10065173/clang_warnings.txt).
-  maybe to automatically improve code style with clang-format (e.g. I saw somewhere tab instead of spaces  e.g.  https://github.com/tesseract-ocr/tesseract/blob/main/src/tesseract.cpp#L424, https://github.com/tesseract-ocr/tesseract/blob/main/src/arch/dotproductsse.cpp#L74... )

From open Issues I would suggest (sorted by priority): 
1. review/merge/close open PR - there are 15...
2. solve [regressions](https://github.com/tesseract-ocr/tesseract/issues?q=is%3Aissue+is%3Aopen+label%3Aregression)
3. if possible [unexpected termination and process hangs bugs](https://github.com/tesseract-ocr/tesseract/issues?q=is%3Aissue+is%3Aopen+label%3A%22unexpected+termination%22%2C%22process+hangs%22+-label%3A%22layout+analysis%22+) should be at least investigated.
4. open issues for [milestone 5.0.0](https://github.com/tesseract-ocr/tesseract/issues?q=is%3Aopen+is%3Aissue+milestone%3A5.0.0) ;-)

**Comment by amitdo** [comment]
Let's set a date:  December 16.

Any issue/PR that is not solved by then, will have to wait to another release.

**Comment by amitdo** [comment]
@stweil, 

Please release Tesseract 5.3.0 this week or next week.

**Comment by zdenop** [comment]
What about to make RC fist - just heads-up from tesseract wrappers?

**Comment by egorpugin** [comment]
Yes, 1-2 weeks RC is better.

**Comment by amitdo** [comment]
We didn't have beta/rc in 5.0.1, 5.1.0 and 5.2.0, but have no objection to have it in 5.3.0.

**Comment by stweil** [comment]
So I'd like to release 5.3.0-rc1 tomorrow and 5.3.0 one week later.

**Comment by amitdo** [comment]
After v5.3.0, should we move to 6.0 or continue with 5.x for some more time?

**Comment by stweil** [comment]
Release [5.3.0](https://github.com/tesseract-ocr/tesseract/releases/tag/5.3.0) is now available.

Thank you for all contributions and support. Merry Christmas!

**Comment by amitdo** [comment]
Should we move to 6.0 or continue with 5.x for some more time?

IMO, we should continue with 5.x (and C++17) for next 6-12 months.

**Comment by egorpugin** [comment]
What is the reason to keep v5?

**Comment by amitdo** [comment]
Hi Egor,

I think we should be conservative and should not change the major release too often.

In general, big rewrites often cause serious regressions. This already happened before.

I also think it's too early to move to C++20.

**Comment by stweil** [comment]
Stabilizing the code and fixing open issues still requires more work. In addition UB Mannheim plans to add more options for `thresholding_method` and PAGE XML output soon. It would also be good to work on reproducable training to replace or enhance the existing models. Switching to C++20 is less urgent and can be postponed until at least 2024. Therefore I'd also prefer to stick to 5.x for the next time.

**Comment by zdenop** [comment]
It would be nice to close as much as possible [issues labelled as "unexpected termination"](https://github.com/tesseract-ocr/tesseract/issues?q=is%3Aissue+is%3Aopen+label%3A%22unexpected+termination%22)

**Comment by amitdo** [comment]
@stweil, I suggest to release 5.3.1 now/next week. Alternatively, set a date, 2-4 weeks from now for that release.

**Comment by stweil** [comment]
Release [5.3.1](https://github.com/tesseract-ocr/tesseract/releases/tag/5.3.1) is now available.

Thank you for all contributions and your support.

**Comment by stweil** [comment]
Plans for next release: several contributions from @JKamlah (see [branches](https://github.com/JKamlah/tesseract/branches)). This will add new functionality (PAGE XML output, NLBIN for image processing, API extension for training, text2image with grayscale support), so it will require a new minor release 5.4.0.

**Comment by stweil** [comment]
Meanwhile several smaller improvements and fixes were added after 5.3.1 which was released 3 months ago, so we could create a bug fix release 5.3.2. If everybody agrees, I can do that on Monday or Tuesday, for example.

**Comment by zdenop** [comment]
ok.

**Comment by stweil** [comment]
That's a valid question. Thanks for the reminder.

Release [5.3.2](https://github.com/tesseract-ocr/tesseract/releases/tag/5.3.2) is now available. Thank you for all contributions and your support.

**Comment by nijel** [comment]
Any plans for a new release? I'd like to see https://github.com/tesseract-ocr/tesseract/commit/d7c071197e771263e923b731af05302a7e68efbd being released.

**Comment by stweil** [comment]
Commit 063ad31a6018af928816ead25ffbb7584f667d22 is even more important. So yes, there are good reasons for a new release, and I'll prepare it as soon as possible.

**Comment by stweil** [comment]
ChangeLog (please comment):

* Small code fixes and improvements to fix Coverity Scan issues.
* Disable -mfpu=neon for aarch64.
* Fix build without git clone in cloned directory (required for FreeBSD).
* Other build fixes for autotools, cmake and sw.
* Fix regression in layout detection which was introduced in release 5.0.0.
* Other small improvements for code and documentation.

Please comment if something should be changed or is missing.

**Comment by amitdo** [comment]
Can you look at issue #4002 and try to resolve it?

If you don't think you'll be able to fix it in short time, you can still release 5.3.3 without a fix for that issue.

**Comment by stweil** [comment]
Thank you for that hint. That was another regression (since 5.0.0-rc2) which is fixed in pull request #4141.

**Comment by stweil** [comment]
Release [5.3.3](https://github.com/tesseract-ocr/tesseract/releases/tag/5.3.3) is now available. Binaries for Windows will be available in a few hours, too.

As always thank you for all contributions (especially from several new contributors) and your support.

**Comment by stweil** [comment]
Should we publish a 5.3.4 this weekend? List of important changes:

- Fixes for scrollview
- Fixes for autoconf, clang and sw builds
- New parameter curl_cookiefile
- Output directory list from "combine_tessdata -d" to stdout

Is there anything missing?

**Comment by amitdo** [comment]
+1 to releasing 5.3.4.

**Comment by amitdo** [comment]
@AlexanderP,

I hope you can push Tesseract 5.3.4 to Debian unstable so it will find its way to Ubuntu 24.04.

**Comment by stweil** [comment]
List of important changes since 5.3.4:

* Fix grey result of indexed PNG in pdfrenderer
* [pdfrenderer] Do not include space symbol into the word pdf box
* [pdfrenderer] Add some debugging capabilities
* normstrngs: add more hyphens and quotes
* Updates for documentation, build and test scripts

As soon as the renaming of frk -> deu_latf (see https://github.com/tesseract-ocr/langdata_lstm/pull/59) is finished, I can publish a new release 5.3.5.

**Comment by stweil** [comment]
Maybe we can also fix some more [issues which were reported by Coverity scan](https://scan.coverity.com/projects/tesseract-ocr).

**Comment by amitdo** [comment]
>[pdfrenderer] Do not include space symbol into the word pdf box

What about macOS Preview?
https://github.com/tesseract-ocr/tesseract/pull/3139#issuecomment-716044792

**Comment by egorpugin** [comment]
Do not test it with macOS Preview. Test with chrome on macos.
I've checked it after fixing one-off patch.
Chrome displays the document in same manner as on win.

Different programs render pdf differently and we do not know if they are correct.
So focus only on the most used render (chrome) for now.

**Comment by amitdo** [comment]
First, this remind me the days when large percent of websites told you: "This site is best viewed by Internet Exploer".

I think the most used pdf render on macOS is Preview. Mac user prefer to use Apple's tools. Same for Chrome vs. Safari.

Also, this patch make Evice selection behave worse than how it behave before.

There is also a suggested alternative patch that might work better across different renderer.

**Comment by egorpugin** [comment]
We have four cases:
1) tess fails, viewer fails
2) tess fails, viewer ok
3) tess ok, viewer fails
4) tess ok, viewer ok

Is chrome viewer ok?
Is macos preview ok?
Is evince ok?
Is adobe viewer ok?

Different viewer behavior means that someone is correct and others are not.
Which one is the correct viewer?

**Comment by amitdo** [comment]
<Different viewer behavior means that someone is correct and others are not.

No. Its means that the pdf format is very complex and the spec itself is not clear even for pdf experts.

Also, all the pdf viewer use its own 'c)ever guesses' techniques  for some features of the format. This is very relevant here.

**Comment by amitdo** [comment]
I don't like this patch, but if @stweil will explicitly say he has no issue with it, I will stop talking about it.

**Comment by amitdo** [comment]
Let's cpntinue the discusion about the pdf renderer in issue #2879.

**Comment by amitdo** [comment]
OK. I decided to remove my objection to the recent changes in the pdf renderer.

**Comment by amitdo** [comment]
@stweil,

What about the useless OpenCL code? It's about time we remove it.

**Comment by jbarlow83** [comment]
@stweil The changes in the PDF renderer are compatible with OCRmyPDF and yield a slight improvement in text positioning on Evince. LGTM.

I tested Tesseract commit 2b07505 which includes egorpugin's changes by examining visual results in Evince using both OCRmyPDF's wrapper around the Tesseract PDF renderer (`--pdf-renderer sandwich`) and the direct output from PDF renderer. Did not check macOS Preview where the trouble usually is. I also confirmed that the PDF produced by debugging changes commit (which could have an impact on production output) is still producing a syntactically valid PDF when debugging is off.

**Comment by amitdo** [comment]
The next release will be 5.4.0.

**Comment by amitdo** [comment]
[**amitdo**](https://github.com/amitdo) commented [Mar 18, 2024](https://github.com/tesseract-ocr/tesseract/issues/3673#issuecomment-2004804558) •

>@stweil,
>
>What about the useless OpenCL code? It's about time we removed it.

Done in #4220.

**Comment by amitdo** [comment]
@stweil,

Can you please release 5.4.0 in the next few days?

**Comment by stweil** [comment]
That's my plan.

**Comment by amitdo** [comment]
on Apr 24

> That's my plan.

So... what's your current plan?

**Comment by stweil** [comment]
Done now, [5.4.0 is available](https://github.com/tesseract-ocr/tesseract/releases/tag/5.4.0). Sorry for the delay. And as always many thanks to all contributors and supporters who helped with issues, discussions and pull requests.

Some things (pull rquests) remain open for follow-up releases.

**Comment by amitdo** [comment]
Thanks[ @stweil]!

https://github.com/tesseract-ocr/tesseract/commit/b8961a7375d54fc4e1a7f8fa2858705f1c13fe84 is not mentioned in the 5.4.0 changes.

**Comment by stweil** [comment]
The list of changes is generated automatically by GitHub which only uses the information from pull request. Therefore direct commits which were made by maintainers might be missing. I updated the release information now with an initial comment, but feel free to suggest further improvements (or change the release notes as required).

**Comment by stweil** [comment]
I think we need a 5.4.1 because of a regression with legacy models (issue #4257) which is now fixed in main. Is there anything else which should be included in the bug fix release?

**Comment by zdenop** [comment]
[GA cmake win64](https://github.com/tesseract-ocr/tesseract/actions/workflows/cmake) build started to crash 4 days ago.  
I am not able to replicate the problem on my laptop with MSVC 2019 (GA should use MSVC 2022).
When I download Artifacts, they crash also for `tesseract --list-langs`... 

Also, I was able to replicate the problem described here https://github.com/tesseract-ocr/tesstrain/issues/394 on Windows with the 5.4.0 code:
```sh
git clone --depth 1 https://github.com/tesseract-ocr/tesstrain

cd tesstrain
mkdir data
unzip ocrd-testset.zip -d data/ocrd-testset-ground-truth

make training MODEL_NAME=ocrd-testset START_MODEL=ces TESSDATA=..../tessdata/best  2>&1 | tee training.log
```
I am checking whether the lastest code fixed also this...

**Comment by stweil** [comment]
I agree that it would be good to clarify these two issues, but I cannot reproduce them up to now.

**Comment by zdenop** [comment]
[GA cmake win64](https://github.com/tesseract-ocr/tesseract/actions/workflows/cmake-win64.yml) seem like GA/Win env issue, so you do not need to wait for this (I already have minimal working version, now I try to add steps to find out real cause of problem)

**Comment by zdenop** [comment]
Seems like `Release` and `RelWithDebInfo` build (Visual Studio 17 2022;  MSVC 19.40.33811.0 / MSC v.1940) is crashing tesseract. `Debug` works fine so I will keep it for GA.
 MSVC 19.39.33523.0 / MSC v.1939 `Release` build [worked fine](https://github.com/tesseract-ocr/tesseract/actions/runs/9392353887/job/25866425030#step:17:37). For me this is the compiler issue, but I have no clue where and how to report it.

**Comment by kloczek** [comment]
5.4.1 has one issue. It uses bundled googletest included in source tree as submodules.
Why not use system installed gtest? 🤔 All distros provides gtest ..

**Comment by amitdo** [comment]
@kloczek,

It's not the right place to discuss the gtest issue.

The gtest issue is not new and you we discussed it in the past in #2838  and #3679.

**Comment by stweil** [comment]
It's time for a new bug fix release. Is there anything urgent which should be included or fixed in the next release?

**Comment by zdenop** [comment]
I am in the process of creating cmake files with autotools (leptonica has it already) This is not critical, but it takes more time than I expect it...

**Comment by stweil** [comment]
... and it currently breaks the autotools builds.

**Comment by zdenop** [comment]
This is unrelated topic as cmake generate tesseract.pc from other template (`tesseract.pc.cmake`) Maybe it could be unified, but this is not topic for now.

**Comment by amitdo** [comment]
@stweil, please go ahead with a new release.

**Comment by stweil** [comment]
I'll try to fix the CI failures before tagging a new release.

**Comment by egorpugin** [comment]
I've checked this issue 
https://github.com/tesseract-ocr/tesseract/pull/4330/files

`TessBaseAPI::GetIterator()` and some other methods (like `GetUTF8Text()`) return raw memory.
It would be nice so they return `unique_ptr<T>` instead.
Doing this we clearly state memory management of returned objects instead of documentation mention.

I propose to impove memory management of public APIs in tess v6 because it is API breakage.

In addition C API implementation will be updated from
```
TessResultIterator *TessBaseAPIGetIterator(TessBaseAPI *handle) {
  return handle->GetIterator();
}
```
to
```
TessResultIterator *TessBaseAPIGetIterator(TessBaseAPI *handle) {
  return handle->GetIterator().release();
}
```

So C API will be retained the same.

---

So,

1. How and when do we want API breaking changes?
2. What other public API/ABI changes do we want? We need a tracking issue for it? Do we have one already?
3. I think we have enough 5.x.x releases already, maybe switch master branch to v6 and create separate v5 branch for small fixes?

**Comment by stweil** [comment]
I just added #4336, and we can discuss and track API changes there.

**Comment by amitdo** [comment]
@stweil, when do you plan to make a new release?

**Comment by stweil** [comment]
Soon (this weekend) unless there is something open or missing which requires more time. The new release will contain enough changes to justify the move to 5.5.0.

**Comment by stweil** [comment]
Meanwhile [release 5.5.0](https://github.com/tesseract-ocr/tesseract/releases/tag/5.5.0) is available. Thank you to everybody who contributed in any way.

**Comment by alfredordgzs** [comment]
Could you please update tesseract to OCR, and calculate ages?

**Comment by stweil** [comment]
I think it's time for a new release 5.5.1, maybe on 2025-05-01.
Are there important things which are still missing for the new release?

**Comment by AlexanderP** [comment]
@egorpugin

Debian Sid

```
gcc -v

Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-linux-gnu/14/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 14.2.0-19' --with-bugurl=file:///usr/share/doc/gcc-14/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,m2,rust --prefix=/usr --with-gcc-major-version-only --program-suffix=-14 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/libexec --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-libstdcxx-backtrace --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --enable-cet --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/reproducible-path/gcc-14-14.2.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/reproducible-path/gcc-14-14.2.0/debian/tmp-gcn/usr --enable-offload-defaulted --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-serialization=3
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 14.2.0 (Debian 14.2.0-19)
```

**Comment by egorpugin** [comment]
@AlexanderP 

Can you try to change this function with the following and see if the error is still there please?
https://github.com/tesseract-ocr/tesseract/blob/main/src/ccutil/tprintf.h#L35

```
template <typename ... Types> auto tprintf(cosnt char *fmt, Types && ... args) {
  return fprintf(get_debugfp(), fmt, std::forward<Types>(args)...);
}
```

**Comment by egorpugin** [comment]
Reverted to va_args https://github.com/tesseract-ocr/tesseract/commit/4a39a4960776a2541c6e818052b029978bc9e191

**Comment by davidecavestro** [comment]
Do you plan to provide an official RPM for 5.5.1?
I could be wrong, but it seems that RPMs from @AlexanderP are no longer available for previous versions.

PS: I just [posted on the group](https://groups.google.com/g/tesseract-ocr/c/SOf6z1iEvfI) yesterday, but I'm also writing here in the hope that Alexander will see it.

**Comment by davidecavestro** [comment]
Thank you very much!
Since I also need the rpm for the old Centos8 (for use on UBI8), would you mind adding them or even sharing the old Centos8 rpm sources so we can manage to build them?
Alternatively, I would start from the sources of the rpms you just released.

**Comment by amitdo** [comment]
@stweil, please make a new release.

**Comment by stweil** [comment]
> [@stweil](https://github.com/stweil), please make a new release.

Yes, I want to do this on next Sunday. And to simplify the release process, I suggest that [ChangeLog](https://github.com/tesseract-ocr/tesseract/blob/main/ChangeLog) will no longer track all changes but only link to the release notes on GitHub. Or should we remove this file?

**Comment by amitdo** [comment]
I agree to the suggested change (link to the release notes or removal of the whole file).

**Comment by stweil** [comment]
The new [release 5.5.1](https://github.com/tesseract-ocr/tesseract/releases/tag/5.5.1) is now available. Thank you to everyone who contributed to it by reporting issues, providing pull requests, testing, offering advice, and participating in discussions.

**Comment by aswanee** [comment]
Arabic Eastern Number and Dates still not readable tell now as (٢٠٢٥/٠٢/٢٣
٢٠١٥/٠١/٢٦
٢٠١٧/١٢/٢٤
)
